stor/obs.py

```diff
@@ -450,6 +452,22 @@ def close(self):
                 self.flush()
             self._buffer.close()
         self.closed = True
+        # we want to wait_on_close on DXPath only if no error was thrown while writing the file
+        if not any(sys.exc_info()):  # pragma: no cover
+            self.dx_close_helper()
+
+    def dx_close_helper(self):
```

Comment:
naming wise a little funky:

probably better to make it private and just call

```
_wait_on_close
```

I'd also suggest changing signature to:

```
self._wait_on_close(20)
```

and then change: `wait_on_close` line as suggested below

Reply:
does this need _wrap_dx_call too ?

Reply:
Changed name to _wait_on_close. And yes, it should have been wrapped in _wrap_dx_call. FIxed.

I am not so sure about changing the signature to 
```
    def _wait_on_close(self):
        if isinstance(self._path, stor.dx.DXPath):
            wait_on_close = wait_on_close if wait_on_close is not None else stor.settings.get()['dx']['wait_on_close']
```
and then call self._wait_on_close(20) from open, because then open would never work with settings. We are basically hardcoding 20 here. e.g. This won't take wait_on_close to be 0:
```
        with stor.settings.use({'dx': {'wait_on_close': 0}}):
            with dx_p.open(mode='wb') as obj:
                obj.write(b'hello world')
```

Let me know if I am getting this wrong.

Reply:
no I'm not suggesting you pass an explicit value to `_wait_on_close`, I'm just suggesting allowing specifying alternate wait period.